src/plugins/recordTypes/collectionobject/fields.js

```diff
@@ -0,0 +1,489 @@
+import { defineMessages } from 'react-intl';
+import { computeDocDisplayName } from '../utils';
+
+export default (pluginContext) => {
+  const {
+    AutocompleteInput,
+    CompoundInput,
+    OptionPickerInput,
+    TextInput,
+    TermPickerInput,
+    CheckboxInput,
+  } = pluginContext.inputComponents;
+
+  const {
+    configKey: config,
+  } = pluginContext.configHelpers;
+
+  const {
+    DATA_TYPE_BOOL,
+  } = pluginContext.dataTypes;
+
+  return {
+    document: {
+      'ns2:collectionobjects_cinefiles': {
+        [config]: {
+          compute: computeDocDisplayName,
+          service: {
+            ns: 'http://collectionspace.org/services/collectionobject/local/cinefiles',
+          },
+        },
+        docTitleArticle: {
+          [config]: {
+            messages: defineMessages({
+              name: {
+                id: 'field.collectionobjects_cinefiles.docTitleArticle.name',
+                defaultMessage: 'Article',
+              },
+            }),
+            view: {
+              type: OptionPickerInput,
+              props: {
+                source: 'articles',
+              },
+            },
+          },
+        },
+        docTitle: {
+          [config]: {
+            messages: defineMessages({
+              name: {
+                id: 'field.collectionobjects_cinefiles.docTitle.name',
+                defaultMessage: 'Name',
+              },
+              fullName: {
+                id: 'field.collectionobjects_cinefiles.docTitle.fullName',
+                defaultMessage: 'Title name',
+              },
+            }),
+            required: true,
+            view: {
+              type: TextInput,
+            },
+          },
+        },
+        docDisplayName: {
+          [config]: {
+            messages: defineMessages({
+              name: {
+                id: 'field.collectionobjects_cinefiles.docDisplayName.name',
+                defaultMessage: 'Display Name',
+              },
+            }),
```

Comment:
Need to configure a searchView, or else the field will also be read only on the search form.